### PR TITLE
Allow choosing the Vulkan max API version when creating Instance

### PR DESCRIPTION
**Pull request by Rua**
* [x] Added an entry to `CHANGELOG_VULKANO.md` or `CHANGELOG_VK_SYS.md` if knowledge of this change could be valuable to users
* [x] Updated documentation to reflect any user-facing changes - in this repository
* [ ] Updated documentation to reflect any user-facing changes - PR to the [guide](https://github.com/vulkano-rs/vulkano-www) that fixes existing documentation invalidated by this PR.
* [x] Ran `cargo fmt` on the changes

This is the followup PR mentioned in #1584, and is another step towards completing #1509. I've added the ability for users to choose the API version themselves, and I added a bunch of documentation about how this works to the `Instance` type. Additionally, I rewrote the `features` module to better support multiple Vulkan versions. Every Vulkan 1.1 and 1.2 feature is included, and I added them to `vk-sys` as well.

### CODE REVIEW

**Comment by Eliah-Lakhin** [comment]
@Rua Works fine. Please fix the change log. Thank you.

File: CHANGELOG_VULKANO.md [hunk 1]
```diff
@@ -9,12 +9,15 @@
...
 - Added `FunctionPointers::api_version` to query the highest supported instance version.
-- Added `Instance::api_version` and `Device::api_version` to return the actual supported Vulkan version. These may differ between instance and device, and be lower than what `FunctionPointers::api_version` and `PhysicalDevice::api_version` return (currently never higher than 1.1, but this may change in the future).
+- Added `Instance::api_version` and `Device::api_version` to return the actual supported Vulkan version. These may differ between instance and device, and be lower than what `FunctionPointers::api_version` and `PhysicalDevice::api_version` return.
+- Added `Instance::max_api_version`, which returns the maximum version that was specified when creating the instance.
 - Fixed the issue when creating a buffer with exportable fd on Linux(see to #1545).
 - The `draw_indirect` and `draw_indexed_indirect` commands on `AutoCommandBufferBuilder` now check the draw count against the `max_draw_indirect_count` limit.
 - Fixed a few documentation errors.
+- Added support for all core Vulkan 1.1 and 1.2 device features.
```
Comment (Eliah-Lakhin) [inline CHANGELOG_VULKANO.md:20]: @Rua This should be moved into vk_sys changelog.

Comment (Rua) [reply CHANGELOG_VULKANO.md:29]: Rather than moving it, I added a separate line to the vk-sys changelog, since it affects both crates.

**Comment by Eliah-Lakhin** [comment]
@Rua Merged :+1:

**Pull request merged**